liquidctl/driver/kraken3.py

```diff
@@ -647,6 +647,12 @@ def _find_winusb_device(self, vid, pid, serial):
                 return True
         return False
 
+    def _get_fw_version(self, clear_reports=True):
+        if clear_reports:
+            self.device.clear_enqueued_reports()
+        self._write([0x10, 0x01])  # firmware info
+        self._read_until({b"\x11\x01": self.parse_firm_info})
+
```

Comment:
Thinking about the API use case (in opposition to the more common CLI use case), I think it would be nice to store the (first) result and make sure that we wont keep fetching the firmware version from the device over and over again.

Reply:
Makes sense, but what if the firmware is upgraded in the meantime? It's an uncommon situation, just wondering.

Reply:
That's a good point. But I think (even though it's probably not guaranteed anywhere) that our HID/USB handles would all become invalid anyway once the device is rebooted into upgrade mode.

Reply:
Yeah, that's what I'm thinking as well, the devices usually disconnect. Let's cache it, then.

Reply:
Yeah, because it's not only that the device reboots, but also that it generally in upgrade mode it generally comes back with completely different VID, PID and other descriptors.